src/pages/home/report/ReportTypingIndicator.js

```diff
@@ -2,80 +2,74 @@ import PropTypes from 'prop-types';
-import withLocalize, {withLocalizePropTypes} from '@components/withLocalize';
-import compose from '@libs/compose';
+import useLocalize from '@hooks/useLocalize';
+import useNetwork from '@hooks/useNetwork';
+import * as PersonalDetailsUtils from '@libs/PersonalDetailsUtils';
+import * as ReportUtils from '@libs/ReportUtils';
 import useThemeStyles from '@styles/useThemeStyles';
-import * as PersonalDetails from '@userActions/PersonalDetails';
 import ONYXKEYS from '@src/ONYXKEYS';
 
 const propTypes = {
     /** Key-value pairs of user accountIDs/logins and whether or not they are typing. Keys are accountIDs or logins. */
     userTypingStatuses: PropTypes.objectOf(PropTypes.bool),
-
-    /** Information about the network */
-    network: networkPropTypes.isRequired,
-
-    ...withLocalizePropTypes,
 };
 
 const defaultProps = {
     userTypingStatuses: {},
 };
 
-function ReportTypingIndicator(props) {
+function ReportTypingIndicator({userTypingStatuses}) {
+    const {translate} = useLocalize();
+    const {isOffline} = useNetwork();
+
     const styles = useThemeStyles();
-    const usersTyping = useMemo(() => _.filter(_.keys(props.userTypingStatuses), (loginOrAccountID) => props.userTypingStatuses[loginOrAccountID]), [props.userTypingStatuses]);
+    const usersTyping = useMemo(() => _.filter(_.keys(userTypingStatuses), (loginOrAccountID) => userTypingStatuses[loginOrAccountID]), [userTypingStatuses]);
+    const firstUserTyping = usersTyping[0];
+
+    const firstUserTypingID = useMemo(
+        () => (firstUserTyping && Number.isNaN(Number(firstUserTyping)) ? PersonalDetailsUtils.getAccountIDsByLogins([firstUserTyping])[0] : firstUserTyping),
+        [firstUserTyping],
+    );
+
     // If we are offline, the user typing statuses are not up-to-date so do not show them
-    if (props.network.isOffline) {
+    if (isOffline || !firstUserTyping) {
         return null;
     }
 
-    const numUsersTyping = _.size(usersTyping);
-
-    // Decide on the Text element that will hold the display based on the number of users that are typing.
-    switch (numUsersTyping) {
-        case 0:
-            return null;
-
-        case 1:
-            return (
-                <TextWithEllipsis
-                    leadingText={PersonalDetails.getDisplayNameForTypingIndicator(usersTyping[0], props.translate('common.someone'))}
-                    trailingText={` ${props.translate('reportTypingIndicator.isTyping')}`}
-                    textStyle={[styles.chatItemComposeSecondaryRowSubText]}
-                    wrapperStyle={[styles.chatItemComposeSecondaryRow, styles.flex1]}
-                    leadingTextParentStyle={styles.chatItemComposeSecondaryRowOffset}
-                />
-            );
+    // If the user is typing on OldDot, firstUserTyping will be a string (the user's login)
+    const firstUserTypingDisplayName = ReportUtils.getDisplayNameForParticipant(firstUserTypingID, false, false) || firstUserTyping;
```

Comment:
I don't think this will work when users don't know each other.
There's case when `Somone is typing...` but with this change, it will never happen

Reply:
I have 2 solutions in mind:
- `getAccountIDsByLogins` accepts 2nd parameter - `shouldGenerateAccountID`. If it's obvious that 1st param is not `login` but `displayName`, no need to generate random accountID but just return 0 or null.
- check displayName if it comes from oldDot by doing email/phone validation

I prefer 1st one since user can still set any random email as display name

Reply:
which one are you not able to reproduce?

Reply:
Here's repro step:
1. on A, create new chat with B
2. A sends any message to B
3. on B, type something
4. Observe chat on A

Reply:
Another scenario:
1. A create new public room
2. Create new account B
3. B joins that room and type something
4. Observe that room on A

Reply:
@situchan but after creating a new chat and sending a message both users see each other's display name - why show 'Someone'?
<img width="1453" alt="Screenshot 2023-12-01 at 17 25 48" src="https://github.com/Expensify/App/assets/28020445/a4ef44bb-9454-450d-bbff-9de9d2e2edf8">

Reply:
<img width="1085" alt="Screenshot 2023-12-01 at 17 27 46" src="https://github.com/Expensify/App/assets/28020445/7467c9d6-a54c-42de-9cc8-b79f9a16f488">

Reply:
> on A, create new chat with B
A sends any message to B
on B, type something
Observe chat on A

Did you exactly follow this step? A should see "Hidden" for B's display name



https://github.com/Expensify/App/assets/108292595/5f44325b-d7aa-4902-8fb6-95271ced3c52

Reply:
yes, I followed these steps and I didn't see Hidden instead of a name 🤔 (Bella Smith)
<img width="1332" alt="Screenshot 2023-12-01 at 17 39 27" src="https://github.com/Expensify/App/assets/28020445/7a1f2c5c-f3ce-4835-b6bf-aa7cd35463bf">

Reply:
> on A, create **new** chat with B

It seems you missed this. They know each other already. As I said, B should be new to A.
To be simple, create new chat with any new email that doesn't have Expensify account

Reply:
Not at all! I created a brand new account for that 😅

Reply:
So here's clear step:
1. A: Create new chat with email which hasn't created Expensify account yet
2. A: Send any message
3. B: Login that email on another device and open that chat
4. B: Type something
5. A: Observe chat

Reply:
could it be bc I created an account for the same address and domain with different content after + sign (I mean [email]+[somegibberish]@callstack.com)?

Reply:
> Not at all! I created a brand new account for that 😅

Can you also check Onyx db and `login exists in personal details?
If exists, it should be bug to report as it's against secure logins policy

Reply:
can you share full video from the start?
Try to test with new accounts for both

Reply:
the account Id wasn't in personalDetailsList
<img width="887" alt="Screenshot 2023-12-01 at 18 09 55" src="https://github.com/Expensify/App/assets/28020445/bd04990c-7e4d-43c2-9a08-d87d0b0c442e">

https://github.com/Expensify/App/assets/28020445/345ddac9-de95-4365-918f-18280b7454bf


https://github.com/Expensify/App/assets/28020445/9deae522-b8d5-4344-b8cb-ea83354ab48d

Reply:
Please create new chat with situchandrashil7+22@gmail.com and share video

Reply:
I think the purpose of hidden logic is to secure sensitive info (`login` in our case)
So if display name not set, show "Hidden"
if display name set, show that display name

Reply:
but for one of them it was set automatically

Reply:
Maybe you logged with google?
Btw you can unset display name in setting

Reply:
I unset it, still the same. I'll get back to it on Monday

Uploading Screen Recording 2023-12-01 at 18.40.35.mp4…